src/test/java/seedu/address/model/location/LocationTest.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.model.location;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.TypicalLocations.DENVER;
+import static seedu.address.testutil.TypicalLocations.NEW_YORK;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.LocationBuilder;
+
+public class LocationTest {
+    /**
```

Comment:
Could add constructor tests eg: Referencing from person/AddressTest, as constructor has requireAllNonNull similar to person.Address
This could be applied to a lot of previously seen files.
Something like this is possible:
```suggestion
    /**
    @Test
    public void constructor_null_throwsNullPointerException() {
        assertThrows(NullPointerException.class, () -> new Location(null));
    }
    
    @Test
    public void constructor_invalidAddress_throwsIllegalArgumentException() {
        String invalidLocation = "";
        assertThrows(IllegalArgumentException.class, () -> new Location(invalidLocation));
    }
```

Reply:
On further review, this addition would be a big increase in requirements for testcasing and is not suitable for a small modification to a completed pull request.
 We should open a new issue  about this on refactoring and improving testcase coverage on Constructors and Null Checks.